site/_posts/2021-07-03-avoid-lockin-insulate-your-application-and-have-backup-plan-excerpt.md

```diff
@@ -0,0 +1,51 @@
+---
+layout: blog-post
+title: Avoid Lock-In by Insulating Your Application and Having a Backup Plan
+description: Identiverse is an annual identity and IAM conference. FusionAuth showed up and took notes so you don't have to.
+author: Cameron Pavey
+image: blogs/takeaways-identiverse-2021/takeaways-from-identiverse-2021-covid-thoughts-fido-and-more-header-image.png
+category: blog
+tags: conference-report
+excerpt_separator: "<!--more-->"
+---
+
+Years ago your team decided to use a third-party authentication system to avoid the time and cost of building one in-house. But now a better option has hit the market and you're wanting to make the switch. Except, hold on, your old system is so deeply ingrained into your organization that you're practically locked-in to your current vendor.
+
+<!--more-->
+
+How can you avoid this? 
+
+Two strategies that work are:
+
+* Insulate Your Application
+* Have a Backup Plan
+
+_This blog post is an excerpt from [Avoiding Authentication System Lock-in](/learn/expert-advice/identity-basics/avoid-lockin/)._
+
+## Insulate your application
+
+When implementing an authentication system (and various other services provided by third-party vendors, such as storage), there is usually an integration involved. This integration requires writing code in your application to interact with the vendor's service.
+
+It is usually a good idea to write this code so that the vendor is insulated away from your application code. Often, vendors will provide an SDK for interfacing with their service. By wrapping the usage of this SDK and abstracting it with a more generic interface, we can have greater control over how the vendor ties into our application.
+
+If the vendor uses Open Standards as described above, this makes things easier. We can build an interface that describes how we would ideally like to interact with our authentication service and then implement this interface with a wrapper around the vendor's SDK. In the future, if things go south and we need to break away from the vendor, we can just re-implement this interface with an integration for a new provider, and with any luck, things will transition smoothly.
```

Comment:
In the future, if things go south and we need to break away from the vendor, we can just re-implement this interface with an integration for a new provider, and with any luck, things will transition smoothly.

This seems like a long sentence